packages/metro/src/ModuleGraph/worker/__tests__/collectDependencies-test.js

```diff
@@ -349,6 +349,42 @@ it('collects imports', () => {
   ]);
 });
 
+describe('optional dependency', () => {
```

Comment:
```suggestion
describe('optional dependencies', () => {
```

Reply:
Can you expand the test cases here to:
* Verify that nested if statements and function bodies in a try-catch block do not count dependencies as optional.
* Verify that the behavior of this extraction is the same as it is on master now with this option disabled.

Reply:
excellent questions...

> Verify that nested if statements and function bodies in a try-catch block do not count dependencies as optional.

How do we know the import/require within the if-statement or function bodies are not intended to be optional if they are within the **lexical** try-block ?  Note dependency added by a function declared outside of try-block, even if it is invoked within a try-block, will not be considered optional. If we are worrying about that developers accidentally included an import within the try block that does not mean to be optional, maybe we should just giving warning messages for all the unresolved optional packages?

> Verify that the behavior of this extraction is the same as it is on master now with this option disabled.

even though I have verified with tests, there is always the possibility for edge cases we haven't thought about... the only sure way is probably to turn off the whole `tryStatement` processing when the flag is off, I will do that...

Reply:
For the first question: see my answer above about changing the extractor.